aws/resource_aws_launch_template.go

```diff
@@ -952,7 +955,7 @@ func readNetworkInterfacesFromConfig(ni map[string]interface{}) *ec2.LaunchTempl
 		networkInterface.Description = aws.String(v)
 	}
 
-	if v, ok := ni["device_index"].(int); ok && v != 0 {
+	if v, ok := ni["device_index"].(int); ok {
```

Comment:
Can you please switch to just this change so we can close #4475 for now?

Reply:
Did it automatically assign `device_index` 0? I believe that's the behavior when its not supplied (auto incrementing from there).

Reply:
No, it was missing completely. DescribeLaunchTemplateVersions returned:
```json
"NetworkInterfaces": [
    {
        "AssociatePublicIpAddress": false,
        "DeleteOnTermination": false,
        "Ipv6AddressCount": 0,
        "SecondaryPrivateIpAddressCount": 0,
        "SubnetId": "subnet-xxxxxxxx"
    }
],
```

Reply:
@bflad Done, sorry about the delay. I haven't had time to check back on this.